django/core/management/commands/listurls.py

```diff
@@ -0,0 +1,275 @@
+import json
+from importlib import import_module
+from io import StringIO
+
+from django.conf import settings
+from django.contrib.admindocs.views import (
+    extract_views_from_urlpatterns,
+    simplify_regex,
+)
+from django.core.management import color
+from django.core.management.base import BaseCommand, CommandError, CommandParser
+from django.utils import termcolors
+
+FORMATS = (
+    "aligned",
+    "table",
+    "verbose",
+    "json",
+    "pretty-json",
+)
+
+
+COLORLESS_FORMATS = (
+    "json",
+    "pretty-json",
+)
+
+
+class Command(BaseCommand):
+    help = "List URL patterns in the project with optional filtering by prefixes."
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.style.HEADER = termcolors.make_style(opts=("bold",))
+        self.style.ROUTE = termcolors.make_style(opts=("bold",))
+        self.style.VIEW_NAME = termcolors.make_style(fg="yellow", opts=("bold",))
+        self.style.NAME = termcolors.make_style(fg="red", opts=("bold",))
+
+    def add_arguments(self, parser: CommandParser):
+        super().add_arguments(parser)
+
+        # Sorting
+        parser.add_argument(
+            "--unsorted",
+            "-u",
+            action="store_true",
+            dest="unsorted",
+            help="Filter URLs by the given prefix[es]",
+        )
+
+        # Prefix
+        parser.add_argument(
+            "--prefix",
+            "-p",
+            dest="prefixes",
+            help="Only list URLs with these prefixes.",
+            nargs="*",
+        )
+
+        # Format
+        parser.add_argument(
+            "--format",
+            "-f",
+            choices=FORMATS,
+            default="aligned",
+            dest="format",
+            help="Formatting style of the output",
+        )
+
+    def handle(self, *args, **options):
+        # Make sure the prefixes are a list
+        prefixes = options["prefixes"]
+        if prefixes:
+            prefixes = prefixes if isinstance(prefixes, list) else [prefixes]
+
+        url_patterns = get_url_patterns(prefixes=prefixes)
+
+        if not url_patterns:
+            raise CommandError("There are no URL patterns that match given prefixes")
+
+        unsorted = options["unsorted"]
+        no_color = options["no_color"]
```

Comment:
I don't see where this is populated.

Reply:
It's coming from `django.core.management.base.BaseCommand`: https://github.com/django/django/blob/4c65aecfe7ebebf2c90f2cccb6d240cd14c1445c/django/core/management/base.py#L350